src/Lorentz/Contracts/BaseDAO/Proposal.hs

```diff
@@ -0,0 +1,682 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | Proposal related entrypoints
+module Lorentz.Contracts.BaseDAO.Proposal
+  ( propose
+  , proposalMetadata
+  , vote
+  , setVotingPeriod
+  , setQuorumThreshold
+  , flush
+  ) where
+
+import Lorentz
+import Lorentz.Contracts.BaseDAO.Management (authorizeAdmin)
+import Lorentz.Contracts.BaseDAO.FA2 (debitFrom, creditTo)
+import Lorentz.Contracts.BaseDAO.Types
+import Lorentz.Contracts.Spec.FA2Interface (TokenId)
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+------------------------------------------------------------------------
+-- Helper
+------------------------------------------------------------------------
+
+checkIfProposalExist
+  :: forall store pm s. (NiceParameter pm, StorageC store pm)
+  => (ProposalKey : store : s) :-> (Proposal pm : s)
+checkIfProposalExist = do
+  stGet #sProposals
+  ifSome nop (failCustom_ #pROPOSAL_NOT_EXIST)
+
+ensureVotingPeriodIsNotOver
+  :: forall store pm s. (NiceParameter pm, StorageC store pm)
+  => (Proposal pm : store : s) :-> s
+ensureVotingPeriodIsNotOver = do
+  toField #pStartDate
+  swap
+  stToField #sVotingPeriod; int
+  -- push (60 * 60 * 24 :: Natural); -- one day
+  -- mul; -- multiply with the voting period
+  add; -- add to the start_date
```

Comment:
AFAIU commented out code above can be removed now. Also `;` looks redundant here.

Reply:
Fixed this.